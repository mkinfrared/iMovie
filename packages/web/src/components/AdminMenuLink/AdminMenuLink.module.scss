@import "src/styles/animations";
@import "src/styles/colors";

.AdminMenuLink {
  color: $white;
  text-decoration: none;
  @extend .colorTransition;

  &:hover {
    color: $secondary-main;
  }

  &.active,
  &.active:hover {
    background-color: $secondary-main;
    color: $white;
  }

  .icon {
    color: inherit;
  }

  .list {
    background-color: inherit !important;
  }
}
