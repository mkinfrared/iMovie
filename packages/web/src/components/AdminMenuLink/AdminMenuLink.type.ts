export interface AdminMenuLinkProps {
  path: string;
  icon: React.ReactNode;
  name: string;
}
