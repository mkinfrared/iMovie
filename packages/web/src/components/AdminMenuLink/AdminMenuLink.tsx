import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import React from "react";
import { Link, useRouteMatch } from "react-router-dom";

import classNames from "utils/classNames";

import css from "./AdminMenuLink.module.scss";
import { AdminMenuLinkProps } from "./AdminMenuLink.type";

const AdminMenuLink = ({ icon, name, path }: AdminMenuLinkProps) => {
  const match = useRouteMatch({ path });

  return (
    <Link
      to={path}
      className={classNames([css.AdminMenuLink, match && css.active])}
    >
      <ListItem button className={css.list}>
        <ListItemIcon className={css.icon}>{icon}</ListItemIcon>
        <ListItemText primary={name} />
      </ListItem>
    </Link>
  );
};

export default React.memo(AdminMenuLink);
