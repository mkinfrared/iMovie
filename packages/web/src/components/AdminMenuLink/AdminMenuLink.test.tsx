import { render } from "@testing-library/react";
import React, { ReactElement } from "react";

import AdminMenuLink from "./AdminMenuLink";

jest.mock("react-router-dom", () => ({
  useRouteMatch: jest.fn(() => true),
  Link: jest.fn(({ children }) => <div>{children}</div>)
}));

describe("<AdminMenuLink />", () => {
  const path = "christmasPoo";
  const name = "Mr. Hankey";
  const icon = <div>icon</div>;

  let Component: ReactElement;

  beforeEach(() => {
    Component = <AdminMenuLink path={path} name={name} icon={icon} />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
