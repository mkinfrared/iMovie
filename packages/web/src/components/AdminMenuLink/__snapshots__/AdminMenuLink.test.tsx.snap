// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminMenuLink /> should match a snapshot 1`] = `
<div>
  <div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root list MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiListItemIcon-root icon"
      >
        <div>
          icon
        </div>
      </div>
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Mr. Hankey
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;
